doc/reference/reference_sql/sql_user_guide.rst

```diff
@@ -755,14 +777,16 @@ for example ``SELECT 1e318 - 1e318;`` is NULL and ``SELECT 1e318 * 0;`` is NULL.
 SQL operations never return the floating-point value -nan,
 although it may exist in data created by Tarantool's NoSQL. In SQL, -nan is treated as NULL.
 
-A string will be converted to a number if it is used with an arithmetic operator and conversion is possible,
+In older Tarantool versions,
+a string would be converted to a numeric if it was used with an arithmetic operator and conversion was possible,
 for example ``'7' + '7'`` = 14.
 And for comparison, ``'7'`` = 7.
-This is called implicit casting. It is applicable for STRINGs and all numeric data types.
+This is called implicit casting. It was applicable for STRINGs and all numeric data types.
+Starting with Tarantool version 2.10, it is no longer supported.
 
 Limitations: (`Issue#2346 <https://github.com/tarantool/tarantool/issues/2346>`_) |br|
 * Some words, for example MATCH and REGEXP, are reserved but are not necessary for current or planned Tarantool versions |br|
```

Comment:
Since these words are reserved, they cannot be used except within quote marks, so yes their use would cause an error. Although it is a good idea to reserve a word if it is going to be used in a future Tarantool version, that is not the case here, or at least it wasn't the case when I wrote that and I know of no plan now. By the way, there are a few words that will be added to the reserved list soon.